### ISSUE TITLE
Improve automatic bracketing in maps to be usable

### ISSUE THREAD

**Issue by paarriagadap** [owid/owid-grapher#3640]
## Description

Improve the automatically generated map brackets to have a meaningful, usable option that don't involve manual processing every time. This is especially critical in large explorers (indicator-based or csv-based) or when we need to generate multiple charts for a Grapher ID-based explorer.

## Expected behaviour

The system should not generate very specific brackets (see pictures below), but rounded, sensible options in the range of the full data and following our [map bracket guidelines](https://www.notion.so/owid/How-to-pick-map-brackets-fca269b396c3470e8d54dd762390ce2a?pvs=4).

## Screenshots

For the indicator share in extreme poverty, this is the default behaviour:
![share-of-population-living-in-extreme-poverty (1)](https://github.com/owid/owid-grapher/assets/63430031/cbf73d96-7f17-40c9-ace2-d002df2f3dab)

We would like in this example that the brackets suggested would be something closer to
![share-of-population-in-extreme-poverty (10)](https://github.com/owid/owid-grapher/assets/63430031/9c202068-2b10-48e8-924d-20fde5bfbd4b)

## Additional context

- Issue reported by [Max](https://owid.slack.com/archives/C0149NKLZAM/p1654601327373289) in 2022 and by [Pablo R](https://owid.slack.com/archives/C46U9LXRR/p1716452104197559) on May 2024.

**Comment by marcelgerber** [comment]
Additional context: [This document outlines the binning strategies that ArcGIS provides](https://pro.arcgis.com/en/pro-app/latest/help/mapping/layer-properties/data-classification-methods.htm) (thanks Fiona!).

**Comment by pabloarosado** [comment]
Thanks for writing this up, @paarriagadap. Creating map brackets is a big pain!

I've created [a separate issue](https://github.com/owid/owid-grapher/issues/3641) with a proposed implementation of how to improve the current tool in the chart admin. The current issue can be about deciding the optimal strategy (which I think can be harder to figure out).

But for explorers, there's no grapher admin tool, and I think we are from having one, so I've created [this separate ETL issue](https://github.com/owid/etl/issues/2683) to have something useful with streamlit.

**Comment by danyx23** [comment]
We talked about this today in data viz triage and yesterday in the data meeting and there is a good chance that @pabloarosado has time in his upcoming cycle to tackle this problem on the python side and come up with a good algorithm that comes close to our guilelines. Once this is done it should not be too much work to reimplement this in grapher.

**Comment by danyx23** [comment]
#3641 described a solution for this issue. We closed that issue because it was a logical duplicate and copied the text into the text body of this issue.

This issue is scheduled for work in a batch cycle in 2025.3. When implementing this, what we want is to move all our guidelines into grapher itself and maintain the minimum configuration necessary to create beautiful map brackets automatically. Manual override of map bins should still be possible.

**Comment by pabloarosado** [comment]
I just had a look at the description of this issue and updated a few things. [MapBracketer](https://github.com/owid/etl/blob/master/apps/wizard/app_pages/map_brackets.py) already has some logic on how to find the optimal set of brackets. I'd be happy to revisit it or create simplified pseudocode.
